Source/ui_unix/controllerconfigdialog.cpp

```diff
@@ -1,329 +1,99 @@
-
-                        }
-                        else if(Rxml.name() == "Value")
-                        {
-                            QString value = ReadElementValue(Rxml);
-                            key = QKeySequence(value.toInt());
-
-                        }
-                        Rxml.readNext();
-                    }
-                    else
-                    {
-                        Rxml.readNext();
-                    }
-                }
-                if (bindingindex != -1 && !key.isEmpty())
-                {
-                    auto currentButtonId = static_cast<PS2::CControllerInfo::BUTTON>(bindingindex);
-                    if(PS2::CControllerInfo::IsAxis(currentButtonId)) {
-                        if (currentButtonId == PS2::CControllerInfo::ANALOG_LEFT_Y || currentButtonId == PS2::CControllerInfo::ANALOG_RIGHT_Y) {//Invert Up/Down
-                            if (m_axis_bindings[bindingindex] == 0)
-                            {
-                                m_axis_bindings[bindingindex]= key[0];
-                            } else {
-                                m_bindings[bindingindex] = std::make_shared<CPH_HidUnix::CSimulatedAxisBinding>(key[0], m_axis_bindings[bindingindex], type);
-                            }
-                        } else {
-                            if (m_axis_bindings[bindingindex] == 0)
-                            {
-                                m_axis_bindings[bindingindex]= key[0];
-                            } else {
-                                m_bindings[bindingindex] = std::make_shared<CPH_HidUnix::CSimulatedAxisBinding>(m_axis_bindings[bindingindex], key[0]);
-                            }
-                        }
-                    } else {
-                        m_bindings[bindingindex] = std::make_shared<CPH_HidUnix::CSimpleBinding>(key[0], type);
-                    }
-                }
-            }
-        }
-        else
-        {
-            Rxml.readNext();
-        }
-    }
-    file.close();
-
-    if (Rxml.hasError())
-    {
-        fprintf(stderr, "Error: Failed to parse file %s: %s\n", filename.toStdString().c_str() ,file.errorString().toStdString().c_str());
-    }
-    else if (file.error() != QFile::NoError)
-    {
-        fprintf(stderr, "Error: Cannot read file %s: %s\n", filename.toStdString().c_str() ,file.errorString().toStdString().c_str());
-    }
-    return m_bindings;
+	for(int i = 0; i < PS2::CControllerInfo::MAX_BUTTONS; ++i)
+	{
+		std::string button(PS2::CControllerInfo::m_buttonName[i]);
+		std::transform(button.begin(), button.end(),button.begin(), ::toupper);
+
+		InputEventSelectionDialog IESD;
+		IESD.Setup(button.c_str(), m_inputManager,
+				   static_cast<PS2::CControllerInfo::BUTTON>(i), m_inputDeviceManager);
+		if(IESD.exec())
+		{
+			m_inputDeviceManager.get()->DisconnectInputEventCallback();
+			std::this_thread::sleep_for(std::chrono::seconds(2));
```

Comment:
I'm currently using this to give the users few seconds to release the recently pressed key, to avoid retriggering of the same key into a different binding.

I don't think it's ideal, so perhaps mapping all the keys' values in their neutral state to compare against before allowing the user to proceed.